components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -112,4 +124,56 @@ function AutomaticPointsTable:parseManualPoints(team, tournamentCount)
 	return manualPoints
 end
 
-return AutomaticPointsTable
\ No newline at end of file
+function AutomaticPointsTable:queryPlacements(tournaments)
+	local queryParams = {
+		limit = 5000,
+		query = 'tournament, participant, placement, extradata'
+	}
+
+	local tournamentNames = Table.mapValues(tournaments,
+		function(tournament)
+			return tournament.name
+		end
+	)
+
+	local tree = ConditionTree(BooleanOperator.any)
+	local columnName = ColumnName('tournament')
+	Table.forEach(tournaments,
+		function(t)
+			tree:add(ConditionNode(columnName, Comparator.eq, t.name))
+		end
+	)
+	local conditions = tree:toString()
+
+	queryParams.conditions = conditions
+	local allQueryResult = mw.ext.LiquipediaDB.lpdb('placement', queryParams)
+
+	local indexedResults = {}
+
+	local tournamentIndices = Table.map(tournaments,
+		function(index, tournament)
+			indexedResults[index] = {}
+			return tournament.name, index
+		end
+	)

```

Comment:
This loop and the following two loops can just be one for loop

Reply:
I'd either have to do multiple loops or nesting to achieve what I want here, can you please hint on how I'd do it in 1 loop?

Reply:
I was using nested loops which lead to an O(n^2) complexity, so I used these loops,
Can't figure out how I'd do this in 1 loop, can you please hint?

Reply:
This, no?

```lua

	local tree = ConditionTree(BooleanOperator.any)
	local columnName = ColumnName('tournament')
	local tournamentIndices = {}
	Table.iter.forEachIndexed(tournaments,
		function(index, t)
			tree:add(ConditionNode(columnName, Comparator.eq, t.name))
			tournamentIndices[t.name] = index
		end
	)
	local conditions = tree:toString()

	queryParams.conditions = conditions
	local allQueryResult = mw.ext.LiquipediaDB.lpdb('placement', queryParams)

	Table.iter.forEach(allQueryResult,
		function(result)
			result.points = tonumber(result.extradata.prizepoints)
			result.securedPoints = tonumber(result.extradata.securedpoints)

			local tournamentIndex = tournamentIndices[result.tournament]
			local tournament = tournaments[tournamentIndex]

			if Table.isEmty(tournament.placements) then
				tournament.placements = {}
			end
			table.insert(tournament.placements, result)
		end
	)

	return tournaments
```

Reply:
Done